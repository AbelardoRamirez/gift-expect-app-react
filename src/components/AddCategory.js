import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {
    const [newCategorie, setNewCategorie] = useState('');


    const handleSubmit = (e) => {
        e.preventDefault();
        if (newCategorie.trim().length > 3) {
            setCategories(cat => [newCategorie, ...cat]);
            setNewCategorie('');

        }
    }

    const handleNewCategorie = (e) => {
        setNewCategorie(e.target.value);

    }

    return (
        <div className="row border my-2 shadow-lg">
            <div className="col">
                <form onSubmit={handleSubmit}>
                    <p>{newCategorie}</p>
                    <input type="text" className="form-control my-2" onChange={handleNewCategorie} value={newCategorie} />
                    <button type='submit' className="btn btn-block btn-dark my-2">
                        Agregar Nuevo Elemento
                </button>
                </form>
            </div>
        </div>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory;