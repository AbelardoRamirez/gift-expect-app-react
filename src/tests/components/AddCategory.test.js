import React from 'react';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';
import '@testing-library/jest-dom';

describe('Pruebas en <AddCategory/>', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    });

    test('Debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        //Para simular eventos no es necesario agregar *onX*
        const value = 'onChange Text';
        input.simulate('change', { target: { value } });
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(value);
    });

    test('NO! debe de postear la informacion con submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault() { } });
        //expect(setCategories).not.toHaveBeenCalled();
        expect(setCategories).not.toHaveBeenCalled();
    });

    test('Debe de llamar el setCategories & limpiar la caja de texto', () => {
        const value = 'onChange TExt';
        wrapper.find('input').simulate('change', { target: { value } });

        wrapper.find('form').simulate('submit', { preventDefault() { } });
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        //expect(setCategories).toBeCalledTimes(2);
        expect(wrapper.find('input').prop('value')).toBe('');
    });


});