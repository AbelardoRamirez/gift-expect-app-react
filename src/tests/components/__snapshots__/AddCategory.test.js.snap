// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory/> Debe de mostrarse correctamente 1`] = `
<div
  className="row border my-2 shadow-lg"
>
  <div
    className="col"
  >
    <form
      onSubmit={[Function]}
    >
      <p />
      <input
        className="form-control my-2"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        className="btn btn-block btn-dark my-2"
        type="submit"
      >
        Agregar Nuevo Elemento
      </button>
    </form>
  </div>
</div>
`;
